xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -1323,6 +1351,71 @@ where
         Ok(Some(msg))
     }
 
+    async fn process_group_message_error_for_fork_detection(
+        &self,
+        provider: &XmtpOpenMlsProvider,
+        message: &GroupMessageV1,
+        error: &GroupMessageProcessingError,
+    ) -> Result<(), GroupMessageProcessingError> {
+        let group_id = message.group_id.clone();
+
+        if let OpenMlsProcessMessage(ProcessMessageError::ValidationError(
+            ValidationError::WrongEpoch,
+        )) = error
+        {
+            let group_epoch = match self.epoch(provider).await {
+                Ok(epoch) => epoch,
+                Err(_) => {
+                    tracing::info!(
+                        "WrongEpoch encountered but group_epoch could not be calculated"
+                    );
+                    return Ok(());
+                }
+            };
+
+            let mls_message_in = match MlsMessageIn::tls_deserialize_exact(&message.data) {
```

Comment:
same thing here, why not just bubble the error up with `?`

If a message deserializes incorrectly it would probably point to issues other than forking, and we would need the context of the error to diagnose, and its valid to return it form this function since it still indicates an error

Reply:
same reason as above, I would say we can follow your advice to print the error in details.